### DISCUSSION TITLE
LengthValidator and empty string on 6.*

### DISCUSSION THREAD

**Discussion by neufman** [symfony/symfony#45288] [Q&A]
Hi ! Since v6, LengthValidator consider empty string as invalid while the doc explains "As with most of the other constraints, null and empty strings are considered valid values. ". Is it a bug or an outdated doc problem ? Thx for your time !

**Comment by alexisLefebvre** [comment]
One way to show this would be to add a test (in this repo) with this example.

**Reply by neufman** [reply]
A test already exists for empty strings : [testEmptyStringIsInvalid](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php) but it check that violations are raised.
Between 5.4 and 6, [testAllowEmptyString](https://github.com/symfony/symfony/blob/v5.4.4/src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php) has been removed with deletion of `allowEmptyString` option.

To match the documentation, `testEmptyStringIsInvalid` method should be renamed to `testEmptyStringIsValid` and  
`(...) ->assertRaised(); `
sould be replaced by  
`->assertNoViolation();`
